tests/testthat/helper.R

```diff
@@ -0,0 +1,85 @@
+if (!requireNamespace("RBackend", quietly = TRUE)) {
+  stop("RBackend is needed to run the tests. Please install it.", call. = FALSE)
+}
+
+if (!requireNamespace("rjson", quietly = TRUE)) {
+  stop("rjson is needed to run the tests. Please install it.", call. = FALSE)
+}
+
+test_it <- function(lst) {
+  if (is.null(lst$DC_TYPE)) lst$DC_TYPE <- "NormalExercise"
+  if (is.null(lst$DC_PEC)) lst$DC_PEC <- ""
+  if (is.null(lst$DC_SOLUTION)) lst$DC_SOLUTION <- ""
+  if (is.null(lst$DC_SCT)) lst$DC_SCT <- ""
+  lst$DC_COMMAND <- "init"
+  output <- rjson::fromJSON(RBackend::execute(rjson::toJSON(lst)))
+  if (any(sapply(output, `[[`, "type") == "error")) {
+    print(output)
+    stop("init failed")
+  }
+  lst$DC_COMMAND <- "submit"
+  rjson::fromJSON(RBackend::execute(rjson::toJSON(lst)))
+}
+
+get_sct_payload <- function(output) {
+  if (any(sapply(output, `[[`, "type") == "error")) {
+    print(output)
+    stop("an error occured")
+  }
+  output[sapply(output, `[[`, "type") == "sct"][[1]]$payload
+}
+
+get_error_payload <- function(output) {
+  if (any(sapply(output, `[[`, "type") == "sct")) {
+    print(output)
+    stop("no error occured")
+  }
+  output[sapply(output, `[[`, "type") == "error"][[1]]$payload
+}
+
+passes <- function(output, mess_patt = NULL) {
```

Comment:
I noticed most of the tests use test_it, passes, and fails, so between the three we could probably modify it so that...

* `test_it` creates a state and runs and SCT, returns some message (or an error)
* `passes` and `fails` checks that output

alternatively, `test_it` could just not do anything, and we could wrap it in testthat's expect expect_error, etc..

or, we could replace `test_it` with a new function, if you want to keep the origin backend test functions as is.